### ISSUE TITLE
Demo example with upperStr inferenced as Promise<_b> instead of Promise<string>

### ISSUE THREAD

**Issue by trusktr** [JSMonk/hegel#217]
In the example

```js
// Hegel will infer "promisify" as "<_q, _c>((_c) => _q) => (_c) => Promise<_q>"
const  promisify  =  fn  =>  arg  =>  Promise.resolve(fn(arg));

// There, Hegel will infer "<_c>(_c) => Promise<_c>"
const id =  promisify(x  => x);

// And "upperStr" will be inferred as "Promise<string>"
const upperStr =  id("It will be inferred").then(str  => str.toUpperCase());

// Finally, "twiceNum" will be inferred as "Promise<number>"
const twicedNum =  id(42).then(num  => num **  2);
```

upprStr is `Promise<_b>` instead of `Promise<string>` in the playground.

If I add the code

```ts
async function main() {
  const s: string = await upperStr
}
```

I get a type error `Type "_b" is incompatible with type "string"`.

**Comment by JSMonk** [comment]
Oh, thank you a lot. We will fix it soon.
Thank you for the issue ^_^.

**Comment by trusktr** [comment]
No prob! :blush:

**Comment by JSMonk** [comment]
Fixed in 0.0.44
